site-src/geps/gep-851.md

```diff
@@ -0,0 +1,88 @@
+# GEP-851: Allow Multiple Certificate Refs per Gateway Listener
+
+* Issue: [#851](https://github.com/kubernetes-sigs/gateway-api/issues/851)
+* Status: Implementable
+
+## TLDR
+
+Replace `CertificateRef` field with an `CertificateRefs` field in Gateway
+Listeners.
+
+## Goals
+
+* Allow certs to be attached for multiple unique Hostnames for the same
+  Listener. This is most helpful when the hostname is empty on the Listener and
+  attached Routes specify multiple hostnames.
+* Enable specifying RSA and ECDSA certs on the same Listener.
+* Improve cert upgrades by allowing new and old certs to be attached to the same
+  Listener.
```

Comment:
These are the two compelling reasons for me.
How horrible would it be to limit to 2 certs per listener (per hostname when hostname specified in listener)?
We can expand the limit in future.

Reply:
I personally think the other reason is also compelling, so would rather start with a higher limit if possible.